ote/src/clj/ote/services/routes.clj

```diff
@@ -137,18 +148,44 @@
            (delete! db ::transit/route {::transit/id id})
            id))))
 
+(defn link-interface
+  "This is a helper function for users. Links created route to given service.
+  Users can add interface by hand but this is much more convenient for them."
+  [db user  {is-linked? :is-linked?
+             service-id :service-id
+             operator-id :operator-id :as form-data}]
+  (let [interface-url (interface-url operator-id)]
+    (authorization/with-transport-operator-check
+      db user operator-id
+      #(do
+         (if (boolean is-linked?)
+           ;delete
+           (specql/delete! db ::t-service/external-interface-description
+                           {::t-service/transport-service-id service-id
+                            ::t-service/external-interface {::t-service/url interface-url}})
+           (specql/insert! db ::t-service/external-interface-description
+                           {::t-service/external-interface {::t-service/description {}
+                                                            ::t-service/url interface-url}
+                            ::t-service/data-content #{:route-and-schedule}
+                            ::t-service/format #{"GTFS"}
+                            ::t-service/license "CC BY 4.0"
```

Comment:
Hard-coding information like this in a delete clause looks a bit scary, but I don't know is there any better approaches?

Reply:
Better option is to add these datas to some .edn file and explain what these are and why. But I don't do it now :)